frontend/src/metabase-lib/lib/queries/utils/date-filters.ts

```diff
@@ -62,11 +74,34 @@ function getDateTimeDimension(filter: any, bucketing?: string | null) {
   return null;
 }
 
+// note: this is probably buggy because we aren't passing `metadata` and `query`
+function getDateTimeDimensionFromMbql(mbql: any, bucketing?: string) {
+  const dimension = Dimension.parseMBQL(mbql);
+  if (dimension) {
+    if (bucketing) {
+      return dimension.withTemporalUnit(bucketing).mbql();
+    } else {
+      return dimension.withoutTemporalBucketing().mbql();
+    }
+  }
+  return mbql;
+}
+
+function getDateTimeFieldRef(field: Field, bucketing?: string) {
```

Comment:
This function and the two above it look suspiciously similar. Will address in a later PR. Tried consolidating, but there are subtle differences that break things in various places.